modules/gapi/test/gapi_gpu_test.cpp

```diff
@@ -202,6 +209,158 @@ TEST(GPU, Symm7x7_test)
+                                        sizeof(uint8_t) * width,
+                                        height };
+
+    // Create texture description
+    D3D11_TEXTURE2D_DESC desc = {};
+    desc.Width = width;
+    desc.Height = height;
+    desc.MipLevels = 1;
+    desc.ArraySize = 1;
+    desc.Format = DXGI_FORMAT_A8_UNORM; // CV_8U
+    desc.SampleDesc.Count = 1;
+    desc.Usage = D3D11_USAGE_DEFAULT;
+    desc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
+    desc.MiscFlags = D3D11_RESOURCE_MISC_SHARED;
+    desc.CPUAccessFlags = 0; // 0 if CPU access is not required
+
+    HRESULT hr;
+    // Create device
+    ID3D11Device* mDevice = nullptr;
+    hr = D3D11CreateDevice(nullptr,
+                           D3D_DRIVER_TYPE_HARDWARE,
+                           nullptr,
+                           0,
+                           nullptr,
+                           0,
+                           D3D11_SDK_VERSION,
+                           &mDevice,
+                           nullptr,
+                           nullptr);
+
+    if(!SUCCEEDED(hr)) {
+        CV_Error(cv::Error::BadCallBack, "D3D11CreateDevice was failed: " +
+                    std::to_string(HRESULT_CODE(hr)));
+    }
+
+    // Create texture2D by test data
+    hr = mDevice->CreateTexture2D(&desc, &initData, &(*texture));
+
+    if(!SUCCEEDED(hr)) {
+        CV_Error(cv::Error::BadCallBack, "CreateTexture2D was failed: " +
+                    std::to_string(HRESULT_CODE(hr)));
+    }
+
+    if(texture == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "CreateTexture2D was failed: texture is NULL");
+    }
+}
+
+TEST(GPU_D3D11, ConvTexture2DtoCLmem)
+{
+    // Create test data ///////////////////////////////////////////////////////
+    const UINT width = 100;
+    const UINT height = 150;
+    ID3D11Texture2D* texture = nullptr;
+    std::vector<uint8_t> test_data(width * height, 0);
+
+    prepareD3D11Texture2D(width, height, test_data, &texture);
+
+    // Convert texture2D to cl_mem_image //////////////////////////////////////
+    cl_mem       image_mem = nullptr;
+    cl_context   context = nullptr;
+    cl_device_id device = nullptr;
+
+    cv::directx::convertFromD3D11Texture2DtoCLMem(texture, &image_mem, &context, &device);
+
+    if (context == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLMem returns empty cl_mem\n");
+    }
+    if (image_mem == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLMem returns empty cl_context\n");
+    }
+    if (device == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLMem returns empty cl_device_id\n");
+    }
+
+    // Move data to host
+    std::vector<uint8_t> out_test_data(width * height, 0);
+    // Target data
+    const size_t sz0[3] = {0, 0, 0};
+    const size_t sz1[3] = {width, height, 1};
+
+    cl_command_queue q = (cl_command_queue)cv::ocl::Queue::getDefault().ptr();
```

Comment:
must belong to the same device & context were returned from conversion function: may be it make sense to return out_queue also

Reply:
Created with Context and Device instead